cross/hoverkite-firmware/src/hoverboard.rs

```diff
@@ -429,4 +430,21 @@ impl Hoverboard {
             shared.motor.position = 0;
         })
     }
+
+    /// Set the motor PWM values directly for testing.
+    pub fn set_motor_pwm_for_test(&mut self, y_percent: u8, b_percent: u8, g_percent: u8) {
+        free(|cs| {
+            // SHARED must have been initialised by the time this is called.
```

Comment:
is this the kind of invariant that can be enforced with a typestate?

Reply:
I don't think so. The issue is that it is an `Option`, because `Shared` requires some run-time initialisation to create it.